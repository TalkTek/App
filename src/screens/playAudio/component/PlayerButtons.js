import React, { Component } from 'react'
import {
  TouchableHighlight,
  View
} from 'react-native'
import FunctionIcon from '../../../components/img/icon/XLIcon'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import analyticAction from '../../../reducer/analytic/analyticAction'

const mapStateToProps = (state) => {
  return {
    playState: state.audio.isPlaying,
    audioName: state.audio.playingAudioStaticInfo.audioName
  }
}

const mapActionToProps = (dispatch) => {
  return {
    ga: bindActionCreators(analyticAction, dispatch)
  }
}

class PlayerButtons extends Component {
  style = {
    flex: 1,
    alignSelf: 'stretch',
    flexDirection: 'row',
    justifyContent: 'space-around'
  }
  
  _buttonGaEvent(type) {
    if (type !== 'playOrPause')
      this.props.ga.gaSetEvent({
        category: 'capsule',
        action: type,
        value: {
          label: this.props.audioName,
          value: 1
        }
      })
  }

  pressEvent(buttonKey, func) {
    this._buttonGaEvent(buttonKey)
    func()
  }

  render() {
    let { playState, data } = this.props
    let buttons = Object.keys(data||{}).map((buttonKey, i) => {
      let button = data[buttonKey]
      return (
        <TouchableHighlight
          key={buttonKey}
          onPress={this.pressEvent.bind(this, buttonKey, button.func)}
          underlayColor="#fff"
        >
          <View>
            <FunctionIcon
              source={button.twoState
                ? (playState
                  ? button.pauseLink : button.playLink
                )
                : button.link
              }
            />
          </View>
        </TouchableHighlight>
      )
    })
    return <View style={this.style}>{buttons}</View>
  }
}

export { PlayerButtons }
export default connect(mapStateToProps, mapActionToProps)(PlayerButtons)
