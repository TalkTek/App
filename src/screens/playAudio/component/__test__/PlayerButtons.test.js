import React from 'react'
import { shallow } from 'enzyme'
import { PlayerButtons } from '../PlayerButtons'
import {TouchableHighlight} from 'react-native'
import renderer from 'react-test-renderer'

describe('playerButton test', () => {
  it('should render <PlayerButtons> Tag', () => {
    shallow(<PlayerButtons />)
  })

  it('should correct show list from data props', () => {
    const data = {
      backward15: {
        twoState: false,
        link: require('../../assets/img/audioElement/forward.png'),
        func: () => {}
      },
      playOrPause: {
        twoState: true,
        playLink: require('../../assets/img/playAudio/play.png'),
        pauseLink: require('../../assets/img/audioElement/pause.png'),
        func: () => {}
      },
    }
    const element = <PlayerButtons data={data} ga={{gaSetEvent: () => {}}} />
    const tree = renderer.create(element)
    const wrapper = shallow(element)
    expect(wrapper.length).toEqual(1)
    expect(tree).toMatchSnapshot()
    // expect(wrapper.children().find('TouchableHighlight').length).toEqual(1)
  })
})
