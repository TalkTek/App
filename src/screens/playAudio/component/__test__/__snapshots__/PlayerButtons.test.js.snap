// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`playerButton test should correct show list from data props 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-around",
    }
  }
>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": 40,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "width": 40,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View>
      <Image
        source={1}
        style={
          Array [
            Object {
              "height": 40,
              "marginBottom": 0,
              "marginLeft": 0,
              "marginRight": 0,
              "marginTop": 0,
              "width": 40,
            },
            undefined,
          ]
        }
      />
    </View>
  </View>
</View>
`;
